decidim-core/app/packs/src/decidim/redesigned_tooltips.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Initialize the tooltips and assign them
+ *
+ * @param {HTMLElement} node target node
+ * @returns {void}
+ */
+export default function(node) {
+  const { firstElementChild: trigger, lastElementChild: tooltip } = node
+
+  const hide = () => {
+    tooltip.classList.add("is-hidden")
+    tooltip.setAttribute("aria-hidden", true)
+  }
+  const show = () => {
+    tooltip.classList.remove("is-hidden")
+    tooltip.setAttribute("aria-hidden", false)
+  }
+
+  // default behaviour
+  hide()
+  trigger.setAttribute("aria-describedby", tooltip.id)
+
+  // keyboard listener is at root-level
+  window.addEventListener("keyup", (event) => event.key === "Escape" && hide())
```

Comment:
These are generally bound to the `keydown` event, if you look e.g. at:
https://github.com/jonathanlevaillant/a11y-dialog-component/blob/6cda0361a93834ff21b8e4b2e7304d0cb1e2d7a7/src/index.js#L175

I think this would be more expected and consistent behavior.